### PR TITLE
Adds wound armor to ALL jumpsuits that were missing it

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Adds wound armor to ALL jumpsuits that were missing it.

Every jumpsuit, by default, has 5 wound armor. However, because for some godforsaken reason armor datums don't use subtypes (seriously, the fuck?), the vast majority of jumpsuits weren't updated to have anything in case they overrode the base armor.

This includes critical oversights such as any nonstandard jumpsuit meant to be armored (Tracksuits, turtlenecks, admin suit, tgmc suit). This is especially critical a problem on nuclear operatives, who face a lot of combat every round and need that wound armor.

Any nonstandard jumpsuit that also protects from departmental hazards doesn't recieve the wound armor, this is seemingly reasonable with, say, medical jumpsuits, but starts to get weird when it includes engineering, botany, the RD, CMO, and CE..

Plasmaman envirosuits also don't by default. This may or may not be on purpose but I added it on them just in case.

Armored jumpsuits that didn't have the wound resistance now have 10, instead of the base 5, since they're meant to be, well, armored. This might also make durathread useful for something (lol, as if)

## Why It's Good For The Game

Consistency. It's very inconsistent that 'wound' armor randomly pops in and out of places ingame. It requires you to think like a space whale to figure out what's the Best Combat Uniform rather than picking what SHOULD be the right choices.

## Changelog

:cl:
fix: Adds wound armor to ALL jumpsuits that were missing it
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
Is there a reason why we can't subtype these all to be `/under/blah`, so all these can inherent the 5 wound armor?
@ZephyrTFA ?

File: code/modules/clothing/under/_under.dm [hunk 1]
```diff
@@ -30,7 +30,7 @@
 	. = ..()
 	if(random_sensor)
 		//make the sensor mode favor higher levels, except coords.
-		sensor_mode = pick(SENSOR_VITALS, SENSOR_VITALS, SENSOR_VITALS, SENSOR_LIVING, SENSOR_LIVING, SENSOR_COORDS, SENSOR_COORDS, SENSOR_OFF)
+		sensor_mode = pick(SENSOR_VITALS = 3, SENSOR_LIVING = 2, SENSOR_COORDS = 2, SENSOR_OFF = 1)
```
Comment (carlarctg) [inline code/modules/clothing/under/_under.dm:33]: Oh i forgot about this change 

```suggestion
		sensor_mode = pick_weight(SENSOR_VITALS = 3, SENSOR_LIVING = 2, SENSOR_COORDS = 2, SENSOR_OFF = 1)
```

Comment (MrMelbert) [reply code/modules/clothing/under/_under.dm]: `pick_weight` doesn't work like that, you need to do `pick_weight(list(...)`.

**Comment by ZephyrTFA** [comment]
> Is there a reason why we can't subtype these all to be `/datum/armor/under/blah`, so all these can inherit the 5 wound armor?
> @ZephyrTFA ?

zero reason not to, and I recommend doing so instead of this.

File: code/modules/clothing/under/_under.dm [hunk 2]
```diff
@@ -30,7 +30,7 @@
 	. = ..()
 	if(random_sensor)
 		//make the sensor mode favor higher levels, except coords.
-		sensor_mode = pick(SENSOR_VITALS, SENSOR_VITALS, SENSOR_VITALS, SENSOR_LIVING, SENSOR_LIVING, SENSOR_COORDS, SENSOR_COORDS, SENSOR_OFF)
+		sensor_mode = pick_weight(SENSOR_VITALS = 3, SENSOR_LIVING = 2, SENSOR_COORDS = 2, SENSOR_OFF = 1)
```
Comment (ZephyrTFA) [inline code/modules/clothing/under/_under.dm:33]: What's going on here

Comment (carlarctg) [reply code/modules/clothing/under/_under.dm:33]: I farted this pr out at 2 am and went to slepe

**Comment by carlarctg** [comment]
> > Is there a reason why we can't subtype these all to be `/datum/armor/under/blah`, so all these can inherit the 5 wound armor?
> > @ZephyrTFA ?
> 
> zero reason not to, and I recommend doing so instead of this.

Yeah i Dunno why it wasn't like that already lol

**Comment by Ryll-Ryll** [comment]
Good change thanks bro

**Comment by ExcessiveUseOfCobblestone** [comment]
dude who made these armor datums

at the very least youd expect say medical jobs to have the same armor throughout with chemists maybe getting extra acid or something, why is each one a unique entry and not under like a department subtype wtfffff

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 3]
```diff
@@ -305,6 +305,11 @@
 	inhand_icon_state = null
 	armor_type = /datum/armor/plasmaman_engineering
 
+/datum/armor/plasmaman_engineering
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/spacesuits/plasmamen.dm:308]: subtype any added plasmamen armor to the job they correspond to and edit the values as appropriate please

File: code/modules/clothing/under/jobs/rnd.dm [hunk 4]
```diff
@@ -108,9 +108,9 @@
 	name = "geneticist's jumpsuit"
 	icon_state = "genetics"
 	inhand_icon_state = "w_suit"
-	armor_type = /datum/armor/rnd_geneticist
+	armor_type = /datum/armor/clothing_under/rnd_geneticist
 
-/datum/armor/rnd_geneticist
+/datum/armor/clothing_under/rnd_geneticist
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/under/jobs/rnd.dm:113]: oh wow look the science jobs have matching armor but they are both unique entries wow

*baby tantrum tenor gif*

**Comment by carlarctg** [comment]
Why it no worky

**Pull request merged**